### ISSUE TITLE
How to use Tensorboard to plot loss on Colab "TPU" with this repo.

### ISSUE THREAD

**Issue by aditya-malte** [google-research/bert#622]

**Comment by aditya-malte** [comment]
I understand there is some limitation with respect to tf.summary with TPUEstimator

**Comment by saberkun** [comment]
TPU works for the V2 summary (tf.contrib.summary)
More examples are in the tpu official models: https://github.com/tensorflow/tpu/blob/master/models/official/resnet/resnet_main.py#L410

**Comment by aditya-malte** [comment]
By V2, do you mean Tensorflow 2.0?

**Comment by jinamshah** [comment]
@aditya-malte were to able to do this? I would also like to plot the TPUEstimator's loss during training on Tensorboard

**Comment by saberkun** [comment]
No, just https://www.tensorflow.org/api_docs/python/tf/contrib/summary
As we pass the loss through the TPUEstimatorSpec returned from model_fn, the loss will be written into summaries by default. You be able to see the loss curve. Please check out https://cloud.google.com/tpu/docs/tensorboard-setup if your are working with gcp.

**Comment by aditya-malte** [comment]
@saberkun, how do I create a new(another) shell instance in colab(as it is given in the link you have attached that we must create another)?
Thanks

**Comment by saberkun** [comment]
I am not sure if your notebook can have a different process. The colab runtime support multiple instances for sure. So, you may just create another colab and access the gs bucket or the directory that store summary and model checkpoints through Tensorboard. Then, in the second shell instance (colab), you can use Tensorboard to visualize the summary.

**Comment by aditya-malte** [comment]
Thanks you @saberkun , will try out the approach you suggested.

**Comment by faizansuhail89** [comment]
Were you able to implement it? @aditya-malte

**Comment by aditya-malte** [comment]
Ahh, well. I tried for a day or two but failed. Also, the work was done. So, no.
